include/scheduler.h

```diff
@@ -42,18 +42,28 @@ namespace detail {
 		void schedule();
 
 	  private:
-		graph_generator& m_ggen;
-		graph_serializer& m_gsrlzr;
-
-		std::queue<scheduler_event> m_available_events;
-		std::queue<scheduler_event> m_in_flight_events;
+		struct event_shutdown {};
+		struct event_task_available {
+			const task* tsk;
+		};
+		struct event_buffer_registered {
+			buffer_id bid;
+			celerity::range<3> range;
+		};
+		using event = std::variant<event_shutdown, event_task_available, event_buffer_registered>;
+
+		std::unique_ptr<graph_generator> m_ggen;
+		std::unique_ptr<graph_serializer> m_gsrlzr;
```

Comment:
The IMHO very terrible name of `gsrlzr` could have benefitted from some drive-by-renaming.

Reply:
I agree. I've renamed to `gser` now, since all "sibling variables" are abbreviated in a similar manner.

IMO for new classes we should confine this kind of abbreviation to local variables and focus on readability for members.